### ISSUE TITLE
Using Gradient Accumulation

### ISSUE THREAD

**Issue by lersouza** [google-research/t5x#707]
Hey guys!

I am about to pretrain a monolingual model using T5X (thank you for this!).

The routine I'll be following is based on ByT5 paper. However, I currently have access to a smaller TPU (v3-8 core), so 2<sup>20</sup> would not fit into its memory. To accomplish similar results, I am thinking of using gradient accumulation, so I can emulate the same batch size used for pertaining the original ByT5.

I couldn't find any documentation about this, but looking through the code, I guess I would have to:

1. Specify BATCH_SIZE = 1024 (2<sup>20</sup>/1024)
2. Specify trainer.num_microbatches = 16

With this, I hope to fit 64 examples in a step (does T5X auto distribute 8 samples per core here?), but update gradients every 16 steps, emulating a 1024 batch size.

Is that correct?

**Comment by agoliaei** [comment]
Hi,
I was wondering about the same thing. Were you able to figure this out?

**Comment by lersouza** [comment]
Hi @agoliaei . Yes, I have used this setting and it seems to work exactly that way. The results were pretty comparable tonight implementation that we have used with gradient accumulation. :)
